### PR TITLE
Added custom/shortenfastaids

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #XXX 

- [x] Added custom/shortenfastaids for https://github.com/nf-core/modules/pull/4984
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by mahesh-panchal** [comment]
Have you checked there are no other modules in nf-core modules that can provide this functionality?

File: modules/nf-core/custom/shortenfastaids/templates/shortenfastaids.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
```
Comment (maxulysse) [inline modules/nf-core/custom/shortenfastaids/templates/shortenfastaids.py:1]: Can we get some LICENSE and authoring info on this script?

**Comment by GallVp** [comment]
Google searches are leading me to custom scripts.

File: modules/nf-core/custom/shortenfastaids/templates/shortenfastaids.py (hunk 1 above)
Comment (GallVp) [reply modules/nf-core/custom/shortenfastaids/templates/shortenfastaids.py:1]: Thank you @maxulysse 

I have added the authoring and LICENSE info.

**Comment by mahesh-panchal** [comment]
What about https://www.nextflow.io/docs/latest/operator.html#splitfasta. Can this operator replace this module in the subworkflow? For example, specifically using the `record: [ id: true]` or perhaps `text` or `desc` are relevant. Also this can work on files too using the `map` operator.

**Comment by GallVp** [comment]
Replaced with Groovy code, see #4984

**Pull request closed without merging**